### PR TITLE
Add rotate support

### PR DESCRIPTION
**Pull request by kulor**
This is just reviving the work done by @MykolaPrymak in https://github.com/mosch/react-avatar-editor/pull/49 as I need rotate functionality.

Screencast:

react-avatar-rotate

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -399,6 +403,35 @@ class AvatarEditor extends React.Component {
...
+        canvas.width = iHeight;
+        canvas.height = iWidth;
+    }
+    context.save();
+    context.translate((canvas.width / 2), (canvas.height / 2));
+    context.rotate((angle * Math.PI / 180));
+    context.translate(-(iWidth / 2), -(iHeight / 2));
+    context.drawImage(imageState.resource, 0, 0);
+    context.restore();
+    this.handleImageReady(canvas);
```
Comment (hu9o) [inline src/index.js]: This call will generate a spurious call to `this.props.onLoadSuccess` whenever the image is rotated.

(haven't tested yet: I may be wrong)

Comment (kulor) [inline src/index.js]: @hu9o right you were, refactored (fad040d) to avoid said callback being fired

**Comment by hu9o** [comment]
There is still an issue that I mentioned it in @MykolaPrymak's PR: when `angle` is different from 0 and another image is dropped, the image is not rotated, although it should.

This could easily be fixed by calling `rotate()` when the image changes.

However, I should point out this issue stems from that the feature is implemented in a _non-React way_. This piece of code in `componentWillReceiveProps()` is symptomatic of the issue:
```javascript
if (this.props.angle != newProps.angle) {
    this.rotate((newProps.angle - this.props.angle));
}
```

**Comment by kuka** [comment]
https://github.com/mosch/react-avatar-editor/pull/136 works with dropped images and keeps rotation angle intact.

**Comment by kuka** [comment]
I believe this can be closed now that #136 is merged.

**Pull request closed without merging**